### PR TITLE
[PM-11136] Convert LoginEmailService email property to state provider

### PR DESCRIPTION
**Pull request by rr-bw**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-11136

## 📔 Objective

With the Auth Browser Refresh work (in progress), we will have a popout option on each page of the Browser Extension. This means that for certain flows, we will need a better way of retaining / persisting the login email to the popout.

Currently, the LoginEmailService has an `email` property that exists as a normal property on the service. This PR converts it to a state provider (in memory), and updates any components where the `getEmail()` and `setEmail()` methods are being used.

- `getEmail()` changes to `loginEmail$`
- `setEmail()` changes to `setLoginEmail()`

### Web

**Showing (a) normal login, (b) normal login with remember email, (c) password hint flow**

https://github.com/user-attachments/assets/2381a72a-58bd-40c0-9854-0a372176f52b

**Showing Login with Device flow**

https://github.com/user-attachments/assets/d207a2a8-c2db-456e-9198-596b23e9d5a9

**Showing Login via SSO with TD - Admin Auth Approval flow**

https://github.com/user-attachments/assets/699f1ba8-bd31-44c1-abd4-a0f49bb9e099

### Extension

**Showing (a) normal login, (b) normal login with remember email, (c) password hint flow**

https://github.com/user-attachments/assets/aafed98a-cf03-433a-bfd9-8ddcbacba9eb

**Showing Login with Device flow**

https://github.com/user-attachments/assets/27d45665-3455-475d-87e0-119528ae38f3

**Showing Login via SSO with TD - Device Approval flow**

https://github.com/user-attachments/assets/4c24ae1a-eb9d-40e5-a410-bfe5f34685a2

### Desktop

**Showing (a) normal login, (b) normal login with remember email, (c) password hint flow**

https://github.com/user-attachments/assets/3c259375-0b01-44ce-9021-0f4f1529ca40

**Showing Login via SSO with TD - Master Password Approval flow**

https://github.com/user-attachments/assets/37ac5b97-7477-487b-8f92-61a3b5c9dd77

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: libs/angular/src/auth/components/login-via-auth-request.component.ts [hunk 1]
```diff
@@ -112,6 +105,13 @@ export class LoginViaAuthRequestComponent
   }
 
   async ngOnInit() {
+    // TODO: I don't know why this is necessary.
+    // Why would the existence of the email depend on the navigation?
+    const navigation = this.router.getCurrentNavigation();
+    if (navigation) {
+      this.email = await firstValueFrom(this.loginEmailService.loginEmail$);
+    }
+
```
Comment (JaredSnider-Bitwarden) [inline libs/angular/src/auth/components/login-via-auth-request.component.ts:114]: I really think we should remove the navigation logic here. `getCurrentNavigation()` returns the current Navigation object when the router is navigating, and null when idle.  The `ngOnInit` will never execute while the router is navigating... Could you do a really quick test to ensure removing the if doesn't have any weird side effects?  I would expect it to be null today in all cases.

File: apps/browser/src/auth/popup/login.component.ts [hunk 2]
```diff
@@ -80,13 +80,15 @@ export class LoginComponent extends BaseLoginComponent {
     };
     super.successRoute = "/tabs/vault";
     this.showPasswordless = flagEnabled("showPasswordless");
+  }
 
+  async ngOnInit(): Promise<void> {
     if (this.showPasswordless) {
-      this.formGroup.controls.email.setValue(this.loginEmailService.getEmail());
+      this.formGroup.controls.email.setValue(
+        await firstValueFrom(this.loginEmailService.loginEmail$),
```
Comment (JaredSnider-Bitwarden) [inline apps/browser/src/auth/popup/login.component.ts:88]: 💡 Nit pick: it is easier to debug when you assign the result of a service call to a variable and then pass it into set value. Just a pattern we should try to keep to.

**Review by JaredSnider-Bitwarden — COMMENTED** [review]
This is looking really good! No major issues so far!

File: libs/angular/src/auth/components/login-via-auth-request.component.ts (hunk 1 above)
Comment (rr-bw) [reply libs/angular/src/auth/components/login-via-auth-request.component.ts]: I removed the nav logic and confirmed that Login with Device flow still works.

File: apps/browser/src/auth/popup/login.component.ts (hunk 2 above)
Comment (rr-bw) [reply apps/browser/src/auth/popup/login.component.ts]: Done! - https://github.com/bitwarden/clients/pull/10624/commits/9a58730c6f8745300ad7753b634f46a432c40ebb

**Review by JaredSnider-Bitwarden — APPROVED** [review]
Great work on this! LGTM! 🚀

**Review by justindbaur — APPROVED** [review]

**Pull request merged**